src/main/java/org/jenkinsci/plugin/Diskcheck.java

```diff
@@ -10,195 +11,180 @@
  * 
  */
+public class Diskcheck extends BuildWrapper {
+
+    public final boolean failOnError;
+
+    public static FilePath lastFileModified(PrintStream log, FilePath dir) throws UnknownHostException, IOException, InterruptedException {
+        log.println("Determining file to delete ...");
+        List<FilePath> files = dir.listDirectories();
+        long lastMod = Long.MAX_VALUE;
+        FilePath choice = null;
+        for (FilePath file : files) {
+            if (file.lastModified() < lastMod) {
+                choice = file;
+                lastMod = file.lastModified();
+            }
+        }
+        return choice;
+    }
+
+    /**
+     * Constructor taking a list of buildsteps to use.
+     *
+     * @param buildstep list of build steps configured in the UI
+     */
+    @DataBoundConstructor
+    public Diskcheck(boolean failOnError) {
+        this.failOnError = failOnError;
+    }
+
+    /**
+     * Overridden setup returns a noop class as we don't want to add anything
+     * here.
+     *
+     * @param build
+     * @param launcher
+     * @param listener
+     * @return noop Environment class
+     */
+    @Override
+    public Environment setUp(AbstractBuild build, Launcher launcher, BuildListener listener) throws IOException, InterruptedException {
+        return new NoopEnv();
+    }
+
+    /**
+     * Overridden precheckout step, this is where we do all the work.
+     *
+     * Checks to make sure we have some buildsteps set, and then calls the
+     * prebuild and perform on all of them.
+     *
+     * @todo handle build steps failure in some sort of reasonable way
+     *
+     * @param build
+     * @param launcher
+     * @param listener
+     */
+    @Override
+    public Descriptor getDescriptor() {
+        return (Descriptor) super.getDescriptor();
+    }
+
+    @Override
+    public void preCheckout(AbstractBuild build, Launcher launcher,
+            BuildListener listener) throws IOException, InterruptedException {
+        PrintStream log = listener.getLogger();
+        // Default value of disk space check is 1GB		
+        int SpaceThreshold = PluginImpl.getInstance().getSpacecheck();
+
+        log.println("Disk space upper threshold is set to :" + SpaceThreshold + "GB");
+        log.println("Checking disk space now ...");
+
+        // Touch workspace so that it is created on first time.
+        if (!build.getWorkspace().exists()) {
+            build.getWorkspace().mkdirs();
+        }
 
+        Node node1 = build.getBuiltOn();
+        Computer Comp = node1.toComputer();
+        String NodeName = build.getBuiltOnStr();
 
-public class Diskcheck extends BuildWrapper {
+        // This new implementation doesn't even touch this, but given I
+        // am not 100% on what it does I have just left it here.
+        if (DiskSpaceMonitor.DESCRIPTOR.get(Comp) == null) {
+            log.println("No slave data available, waiting and retrying ...");
+            Thread.sleep(10000);
```

Comment:
Please do not use synchronous delays in the main build precess. Makes sense to Asynchronous implementation with Future<>